src/macros.cr

```diff
@@ -76,7 +76,7 @@ macro record(name, *properties)
                    }})
     end
 
-    {{yield}}
+    {{ block.body unless block.is_a?(Nop) }}
```

Comment:
This having to change feels like a smell.

Reply:
I'm not sure I understand the reason, but in my mind `yield` and `block.body` should have exactly the same effect. If they don't then it's pretty confusing, and it's going to be really hard to explain and understand the difference between the two (right now I don't understand the difference)

Reply:
Could we maybe keep the old behavior when there's no yield arguments? In that case there won't be any new variables, and there's no need to stringify and reparse the code.